galley/pkg/runtime/resource/resource.go

```diff
@@ -45,16 +45,30 @@ type Key struct {
 	FullName FullName
 }
 
-// VersionedKey uniquely identifies a snapshot of a config resource in the config space, at a given
-// time.
+
+// VersionedKey uniquely identifies a snapshot of a config resource in the config space.
 type VersionedKey struct {
 	Key
 	Version    Version
 	CreateTime time.Time
 }
 
+// Labels are a map of string keys and values that can be used to organize and categorize
+// resources within a collection.
+type Labels map[string]string
+
+// Annotations are a map of string keys and values that can be used by source and sink to communicate
+// arbitrary metadata about this resource.
+type Annotations map[string]string
+
+type Metadata struct {
```

Comment:
Can we add/move `CreateTime` into the metadata? It shouldn't be part of the versioned key IMO.

Reply:
done.